spring-integration-azure/src/main/java/com/microsoft/azure/spring/integration/eventhub/EventHubTemplate.java

```diff
@@ -126,59 +123,50 @@ public boolean subscribe(String destination, Consumer<EventData> consumer, Strin
     public boolean unsubscribe(String destination, String consumerGroup) {
         Tuple<String, String> nameAndConsumerGroup = Tuple.of(destination, consumerGroup);
 
-        if (!consumerByNameAndConsumerGroup.containsKey(nameAndConsumerGroup)) {
+        if (!processorHostsByNameAndConsumerGroup.containsKey(nameAndConsumerGroup)) {
             return false;
         }
 
-        consumerByNameAndConsumerGroup.remove(nameAndConsumerGroup);
         processorHostsByNameAndConsumerGroup.remove(nameAndConsumerGroup).unregisterEventProcessor();
 
         return true;
     }
 
-    protected EventData toEventData(Message<?> message) {
-        Object payload = message.getPayload();
-        if (payload instanceof EventData) {
-            return (EventData) payload;
-        }
+    private class EventHubProcessor<T> implements IEventProcessor {
 
-        if (payload instanceof String) {
-            return EventData.create(((String) payload).getBytes(Charset.defaultCharset()));
-        }
+        private final Consumer<Message<T>> consumer;
+        private final Class payloadType;
 
-        if (payload instanceof byte[]) {
-            return EventData.create((byte[]) payload);
-        }
-
-        return EventData.create((byte[]) this.messageConverter.fromMessage(message, byte[].class));
-    }
-
-    private class EventHubProcessor implements IEventProcessor {
-
-        private final Tuple<String, String> nameAndConsumerGroup;
-
-        EventHubProcessor(Tuple<String, String> nameAndConsumerGroup) {
-            this.nameAndConsumerGroup = nameAndConsumerGroup;
+        EventHubProcessor(@NonNull Consumer<Message<T>> consumer, @NonNull Class<T> payloadType) {
+            this.consumer = consumer;
+            this.payloadType = payloadType;
         }
 
         @Override
         public void onOpen(PartitionContext context) throws Exception {
             LOGGER.info("Partition {} is opening", context.getPartitionId());
-            checkpointersByNameAndConsumerGroup.putIfAbsent(nameAndConsumerGroup, new EventHubCheckpointer());
-            checkpointersByNameAndConsumerGroup.get(nameAndConsumerGroup).addPartitionContext(context);
         }
 
         @Override
         public void onClose(PartitionContext context, CloseReason reason) throws Exception {
             LOGGER.info("Partition {} is closing for reason {}", context.getPartitionId(), reason);
-            checkpointersByNameAndConsumerGroup.get(nameAndConsumerGroup).removePartitionContext(context);
         }
 
         @Override
+        @SuppressWarnings("unchecked")
```

Comment:
suggest to put `unchecked` to the line instead of whole method.

Reply:
I prefer not letting this annotation distract the main logic. Since this method is short, keep it in the method level for now.